bgls/utils.py

```diff
@@ -100,28 +105,30 @@ def apply_near_clifford_gate(
     """
     if cirq.has_stabilizer_effect(op):
         cirq.protocols.act_on(op, state)
-    else:
-        # assuming T gate
-        # assert isinstance(op,
-        #                   cirq.ops.common_gates.ZPowGate) and \
-        #        op.gate.exponent == 0.25
+    elif isinstance(op.gate, cirq.ops.common_gates.ZPowGate):
+        # note this includes Rz gates
+        theta = np.pi * op.gate.exponent
+        # normalizing s.t. Rz gates have 0 relative phase
+        phase = np.exp((0 + 1j) * theta * (0.5 - op.gate.global_shift))
         probs = np.power(
             np.abs(
                 [
-                    np.cos(np.pi / 8) - np.sin(np.pi / 8),
+                    np.cos(theta / 2) - np.sin(theta / 2),
                     np.sqrt(2.0)
                     * np.exp(-(0 + 1j) * np.pi / 4)
-                    * np.sin(np.pi / 8),
+                    * np.sin(theta / 2),
                 ]
             ),
             2,
         )
         cand_gates = [cirq.I(op.qubits[0]), cirq.S(op.qubits[0])]
         chosen_gate = rng.choice(cand_gates, p=probs / sum(probs))
         cirq.protocols.act_on(chosen_gate, state)
+    else:
+        raise TypeError("Gate is not a z rotation gate!")
```

Comment:
Very confusing error message as seeing this would make you think only z rotation gates are accepted by the function, which is not true. Also, it should be a `ValueError` since I can still pass in an object of type `cirq.Operation` and get this error message.

So, the error message should say something like

```python
raise ValueError(f"Only Clifford and Rz operations are supported, but received operation of type {type(op)}.")
```

Reply:
Updated this! str(op.gate) Prints out what gate it is which is nice